lib/collection/url.js

```diff
@@ -238,81 +238,116 @@ _.assign(Url.prototype, /** @lends Url.prototype */ {
     /**
      * Returns the request path, with a leading '/'.
      *
-     * @param {Object=} options
-     * @param {Object} options.unresolved If set to true, path variables will not be processed
+     * @param {?Booleamn=} [unresolved=false]
      * @returns {string}
+     *
+     * @note deprecated variant in v3.4:
+     *  - param {Object} options
+     *  - param {Object} options.unresolved If set to true, path variables will not be processed
      */
-    getPath: function (options) {
+    getPath: function (unresolved) {
+        if (typeof unresolved === 'object') { // @todo discontinue in v4
+            unresolved = unresolved.unresolved;
+        }
+
+        // for unresolved case, this is super simple as that is how raw data is stored
+        if (unresolved) {
+            return PATH_SEPARATOR + this.path.join(PATH_SEPARATOR);
+        }
+
         var self = this,
-            // eslint-disable-next-line max-len
-            segmentArray = (options && options.unresolved) ? this.path : _.transform(this.path, function (res, segment) {
-                var variable;
-                // check if the segment has path variable prefix followed by the variable name.
-                if (_.startsWith(segment, PATH_VARIABLE_IDENTIFIER) && segment !== PATH_VARIABLE_IDENTIFIER) {
-                    variable = self.variables.one(segment.slice(1)); // remove path variable prefix.
-                }
-
-                variable = variable && variable.valueOf && variable.valueOf();
-                res.push(_.isString(variable) ? variable : segment);
-            }, []),
-            path = segmentArray.join(PATH_SEPARATOR);
-        return PATH_SEPARATOR + path; // add leading slash
+            segments;
+
+        segments = _.transform(this.path, function (res, segment) {
+            var variable;
+
+            // check if the segment has path variable prefix followed by the variable name.
+            if (_.startsWith(segment, PATH_VARIABLE_IDENTIFIER) && segment !== PATH_VARIABLE_IDENTIFIER) {
+                variable = self.variables.one(segment.slice(1)); // remove path variable prefix.
+            }
+
+            variable = variable && variable.valueOf && variable.valueOf();
```

Comment:
Shift the assignment to the if block. `variable` would be undefined in case the `if` block was not hit.